src/dune_rpc_impl/diagnostics.ml

```diff
@@ -52,10 +56,28 @@ let diagnostic_of_error : Build_system.Error.t -> Dune_rpc_private.Diagnostic.t
       ; loc = make_loc (Option.value_exn related.loc)
       })
   in
+  let message =
```

Comment:
Here we add in the hints the same way `User_message.pp` does it. Maybe it would be better to introduce a `hints` field in the RPC type?

Reply:
After thinking about it, adding a hints field is kind of useless so I am happy with the way this is at the moment.